services/cd-service/pkg/repository/repository.go

```diff
@@ -369,10 +370,16 @@ func (r *repository) Apply(ctx context.Context, transformers ...Transformer) err
 }
 
 func (r *repository) Push(ctx context.Context, pushAction func() error) error {
+
+	span, _ := tracer.StartSpanFromContext(ctx, "Apply")
+	defer span.Finish()
+
 	eb := backoff.NewExponentialBackOff()
 	eb.MaxElapsedTime = 7 * time.Second
 	return backoff.Retry(
 		func() error {
+			span, _ := tracer.StartSpanFromContext(ctx, fmt.Sprintf("try to Push at: %v", eb.GetElapsedTime()))
```

Comment:
The action should never be dynamic. If you want to add dynamic data to the span, please use `SetTag` like this:

```
span, _ := tracer.StartSpanFromContext(ctx, "push")
span.SetTag("elapsedTime", eb.GetElapsedTime())
```